ClientRequirments.md

```diff
@@ -0,0 +1,39 @@
+# Platform Requirements for Fluid Framework Client Packages
+
+These are the platform requirements for the current version of Fluid Framework Client Packages.
+These requirements err on the side of being too strict since within a major version they can be relaxed over time, but not made stricter.
+For Long Term Support (LTS) versions this can require supporting these platforms for several years.
+
+It is likely that other configurations will work, but they are not supported: if they stop working, that is not a breaking change.
+If you would benefit from support for something not listed here, file an issue and the product team will evaluate your request.
+When making such a request please include if the configuration already works (and thus the request is just that it becomes officially supported), or if changes are required to get it working.
+
+## Supported Runtimes
+
+-   NodeJs ^20.10.0 except that we will drop support for it [when NodeJs 20 loses its upstream support on 2026-04-30](https://github.com/nodejs/release#release-schedule), and will support a newer LTS version of NodeJS (22) at least 1 year before 20 is end-of-life. This same policy applies to NodeJS 22 when it is end of life (2027-04-30).
+-   Modern browsers supporting the es2022 standard library: in response to asks we can add explicit support for using babel to polyfill to target specific standards or runtimes (meaning we can avoid/remove use of things that don't polyfill robustly, but otherwise target modern standards).
+
+## Supported Tools
+
+-   TypeScript 5.4:
```

Comment:
Seems aggressive since we don't use 5.4 ourselves.

Reply:
A large part of why we want to document these requirements is to allow us to move forward with newer tooling (ex: node 20, updated typescript). Since currently we effectively require our users to have TypeScript as new as we use (since we don't have back compat testing for older TS), I think it makes sense to update our requirements before we update what we use so this being newer than what we use should not be surprising.

That said, it is a very aggressive target: I won't be surprised if a customer asks us to add support for an older version, but until that happens, we might as well target something aggressive just in case its actually ok for us to target 5.4 since getting stuck on 5.2 for the next 3 years if no customer actually needed us to do that would be a waste. We can always add more support for old versions, but we can't move our minimum version forward without a breaking change.